Add API documentation for CLUSTER LINKS command (#1710)
redis/redis-doc: commands/cluster-links.md

> Each node in a Redis Cluster maintains a pair of long-lived TCP links with each
> peer in the cluster, one established toward the peer, another accepted from the
> peer.

Reviewer: New style is to avoid added line breaks, since markdown will remove them anyways. Also, although what you said is technically true about inbound/outbound, semantically they are for inbound/outbound.
```suggestion
Each node in a Redis Cluster maintains a pair of long-lived TCP link with each peer in the cluster: One for sending outbound messages towards the peer and one for receiving inbound messages from the peer.
```

> `CLUSTER LINKS` outputs information of all such peer links as an array, where each
> array element is a RESP3 map that contains various attributes and their values of
> an individual link.

Reviewer: Maps aren't a native data type in RESP2, which is why it shows up as an array below, so perhaps just say:
```suggestion
`CLUSTER LINKS` outputs information of all such peer links as an array, where each array element is a map that contains attributes and their values for an individual link.
```

> The following is an example output:

Reviewer: ```suggestion
@examples

The following is an example output:
```

> ```
> 1)  1) "direction"

Reviewer: ```suggestion
> CLUSTER LINKS
1)  1) "direction"
```

    5) "ctime"
    6) (integer) 1639442739375
    7) "events"
    8) "rw"
    9) "sbufalloc"
   10) (integer) 4512
   11) "sbufused"
   12) (integer) 0
2)  1) "direction"
    2) "from"
    3) "node"
    4) "8149d745fa551e40764fecaf7cab9dbdf6b659ae"
    5) "ctime"
    6) (integer) 1639442739411
    7) "events"
    8) "r"
    9) "sbufalloc"
   10) (integer) 0
   11) "sbufused"
   12) (integer) 0
```

Each RESP3 map is composed of the following attributes of the corresponding cluster link and their values:

1. `direction`: This link is established by the local node `to` the peer, or accepted by the local node `from` the peer.
2. `node`: The node id of the peer.
3. `ctime`: Creation time of the link. (In the case of a `to` link, this is the time when the TCP link is created by the local node, not the time when it is actually established.)
4. `events`: Events currently registered for the link. `r` means readable event, `w` means writable event.
5. `sbufalloc`: Allocated size of the link's send buffer, which is used to buffer outgoing messages toward the peer.
6. `sbufused`: Size of the portion of the link's send buffer that is currently holding data(messages).

@return

> @bulk-string-reply: An array of RESP3 maps where each map contains various attributes and their values of a cluster link.

Reviewer: Not a bulk reply string, it's an array of maps. You can see https://github.com/redis/redis-doc/blob/master/commands/acl-getuser.md for another example of a similar response.
```suggestion
@array-reply: An array of maps where each map contains various attributes and their values of a cluster link.
```
